huaweicloud/services/dms/resource_huaweicloud_dms_kafka_smart_connect_task.go

```diff
@@ -0,0 +1,388 @@
+			"destination_type": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the destination type of the smart connect task.`,
+			},
+			"obs_destination_descriptor": {
+				Type:        schema.TypeList,
+				Elem:        obsDestinationDescriptorSchema(),
+				MaxItems:    1,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the destination parameters of the smart connect task.`,
+			},
+			"topics": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the topics of the smart connect task.`,
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the status of the smart connect task.`,
+			},
+			"created_at": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: `Indicates the creation time of the smart connect task.`,
+			},
+		},
+	}
+}
+
+func obsDestinationDescriptorSchema() *schema.Resource {
+	sc := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"access_key": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the access key of the smart connect task.`,
+			},
+			"secret_key": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Sensitive:   true,
+				Description: `Specifies the secret key of the smart connect task.`,
+			},
+			"consumer_strategy": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the consumer strategy of the smart connect task.`,
+			},
+			"deliver_time_interval": {
+				Type:        schema.TypeInt,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the deliver time interval of the smart connect task.`,
+			},
+			"obs_bucket_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the obs bucket name of the smart connect task.`,
+			},
+			"partition_format": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the partiton format of the smart connect task.`,
+			},
+			"obs_path": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: `Specifies the obs path of the smart connect task.`,
+			},
+			"destination_file_type": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "TEXT",
+				Description: `Specifies the destination file type of the smart connect task.`,
+			},
+			"record_delimiter": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "\n",
+				Description: `Specifies the record delimiter of the smart connect task.`,
+			},
+			"topics": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `Specifies the topics of the task.`,
+			},
```

Comment:
```suggestion
			"topics": {
				Type:        schema.TypeString,
				Optional:    true,
				Description: `Specifies the topics to be dumped.`,
				ExactlyOneOf: []string{"topics","topics_regex"},
			},
```
and fix other places

Reply:
done